File: src/main/java/br/com/blz/testjava/mapper/WarehouseMapper.java
package br.com.blz.testjava.mapper;

import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

import static org.apache.commons.collections4.CollectionUtils.emptyIfNull;

import br.com.blz.testjava.model.Warehouse;
import br.com.blz.testjava.model.dto.WarehouseDTO;

public class WarehouseMapper {

	 public static List<Warehouse> toEntities(List<WarehouseDTO> warehouseDTO) {
	        return  emptyIfNull(warehouseDTO)
	        		.stream()
	                .map(WarehouseMapper::toEntity)
	                .filter(Objects::nonNull)
	                .collect(Collectors.toList());
	 }
	 
	 
	 public static Warehouse toEntity(WarehouseDTO warehouseDTO) {
		 Warehouse warehouse = null;
 
		 warehouse = Warehouse.builder()
                    .locality(warehouseDTO.getLocality())
                    .quantity(warehouseDTO.getQuantity())
                    .type(warehouseDTO.getType())
                    .build();

	        return warehouse;
	    }
}

File: src/main/kotlin/br/com/blz/testjava/warehouse/mapper/WarehouseMapper.kt
package br.com.blz.testjava.warehouse.mapper

import br.com.blz.testjava.warehouse.dto.WarehouseDTO
import br.com.blz.testjava.warehouse.model.Warehouse

class WarehouseMapper {
  companion object {

    fun toEntity(warehouseDTO: WarehouseDTO): Warehouse {
      return Warehouse(warehouseDTO.locality, warehouseDTO.quantity, warehouseDTO.type)
    }

    fun toDto(warehouse: Warehouse): WarehouseDTO {
      return WarehouseDTO(warehouse.locality, warehouse.quantity, warehouse.type)
    }
  }
}
